build-tools/DEV.md

```diff
@@ -0,0 +1,162 @@
+
+#### Type Compatibility Issues
+
+12. **typescript** - Pinned to `~5.4.5`
+    - Latest: `~5.7.x`
+    - Issue: Version 5.9+ has stricter type checking that exposes issues with @octokit dependencies
+    - Error: `Cannot find name 'ErrorOptions'`
+    - Used in: `build-cli` (devDependency)
+
+13. **ts-morph** - Pinned to `^22.x`
+    - Latest: `^24.x`
+    - Issue: Version 27+ requires newer TypeScript lib types
+    - Error: `Cannot find name 'MapIterator'`
+    - Used in: `build-cli`
+
+14. **azure-devops-node-api** - Pinned to `^11.x`
+    - Latest: `^15.x`
+    - Issue: Version 15 has incompatible type definitions
+    - Error: `Types have separate declarations of a private property` and `Type is missing properties`
+    - Used in: `build-cli`, `bundle-size-tools`
+
+#### API/Structure Breaking Changes
+
+15. **eslint** - Pinned to `~8.57.0`
+    - Latest: `~9.x`
+    - Issue: Version 9 uses flat config system incompatible with existing configuration
+    - Error: `ESLint configuration is invalid: Unexpected top-level property "__esModule"`
+    - Used in: `build-cli` (devDependency)
+
+16. **eslint-config-oclif** - Pinned to `^5.x`
+    - Latest: `^6.x`
+    - Issue: Version 6 requires ESLint 9
+    - Used in: `build-cli`, `version-tools` (devDependency)
+
+17. **@fluidframework/eslint-config-fluid** - Pinned to `^6.x`
+    - Latest: `^8.x`
+    - Issue: Version 8 requires ESLint 9
+    - Used in: `build-cli`, `build-infrastructure`, `build-tools`, `version-tools` (devDependency)
+
+18. **npm-check-updates** - Pinned to `^16.x`
+    - Latest: `^19.x`
+    - Highest compatible: `^16.x` (v17+ changed internal module structure)
+    - Issue: Version 17+ changed internal module structure and removed exported types
+    - Error: `Cannot find module 'npm-check-updates/build/src/types/IndexType.js'` and type errors
+    - Used in: `build-cli`
+
+### Intermediate Version Upgrade Opportunities
+
+The following package was successfully upgraded to an intermediate version:
+
+- **date-fns**: Upgraded from `^2.x` → `^3.x` ✅ (v4+ is ESM-only)
+
+Note: execa cannot be upgraded beyond v5 despite v6-v7 being CommonJS-compatible, because they removed default export compatibility that breaks TypeScript module resolution in build-infrastructure.
+
+### Migration Path
+
+To upgrade these dependencies, the build-tools codebase would need to be migrated from CommonJS to ESM. This is a significant undertaking that would involve:
+
+1. Converting all TypeScript source files to use ESM imports/exports
+2. Updating `package.json` to include `"type": "module"`
+3. Changing file extensions or configuring TypeScript to output ESM
+4. Updating all import statements to include file extensions where required by ESM
+5. Handling any CommonJS-only dependencies that don't have ESM equivalents
+6. Updating the build and test infrastructure
+
+Until this migration is completed, these dependencies must remain at their current major versions.
```

Comment:
Remove this section. We just want to document the current state, not make claims about the future.